pylot/drivers/carla_traffic_light_invasion_sensor_operator.py

```diff
@@ -124,17 +112,23 @@ def on_pose_update(self, msg):
                     seg2 = (lft_lane_wp, rgt_lane_wp)
                     if self.is_vehicle_crossing_line(seg1, seg2):
                         location = traffic_light.get_transform().location
-                        message = TrafficInfractionMessage(
+                        message = TrafficInfractionMessageTuple(
                             TrafficInfractionType.RED_LIGHT_INVASION,
                             pylot.utils.Location.from_simulator_location(
-                                location), msg.timestamp)
-                        self._traffic_light_invasion_stream.send(message)
+                                location))
+                        context.write_stream.send(
+                            erdos.Message(context.timestamp, message))
                         self._last_red_light_id = traffic_light.id
                         break
 
-    def run(self):
+    def run(
+        self,
+        read_stream: ReadStream,
+        left_write_stream: WriteStream,
+        right_write_stream: WriteStream,
```

Comment:
Should be 2 read streams, 1 write stream. Also, could you rename the read streams to `vehicle_id_stream` and `pose_stream`?